File: pt-python/examples/observability.py

from aws_lambda_powertools import Logger, Metrics, Tracer
from aws_lambda_powertools.logging import correlation_paths
from aws_lambda_powertools.metrics import MetricUnit
from aws_lambda_powertools.utilities.typing import LambdaContext

logger = Logger()
metrics = Metrics()
tracer = Tracer()

@metrics.log_metrics(capture_cold_start_metric=True)
@tracer.capture_lambda_handler
@logger.inject_lambda_context(log_event=True)
def lambda_handler(event: dict, context: LambdaContext):
  metrics.add_metric(name="CartCheckedOut", unit=MetricUnit.Count, value=1)
  
  cart_state = event.cart_state
  
  tracer.put_annotation(key="CusomderId", value=cart_state.customer_id)
  metrics.add_metadata(key="CartDetails", value=cart_state)
  
  # ... rest of the code

File: typescript/src/observability.ts
import type { LambdaInterface } from "@aws-lambda-powertools/commons/types";
import { Logger } from "@aws-lambda-powertools/logger";
import { Metrics, MetricUnit } from "@aws-lambda-powertools/metrics";
import { Tracer } from "@aws-lambda-powertools/tracer";
import type { Context } from "aws-lambda";
import type { CartCheckedOutEvent } from "./types";

const logger = new Logger();
const metrics = new Metrics();
const tracer = new Tracer();


class Lambda implements LambdaInterface {

  @metrics.logMetrics({ captureColdStartMetric: true })
  @tracer.captureLambdaHandler()
  @logger.injectLambdaContext({ logEvent: true })
  public async handler(event: CartCheckedOutEvent, _context: Context): Promise<void> {
    metrics.addMetric('CartCheckedOut', MetricUnit.Count, 1);

    tracer.putAnnotation('CustomerId', event.customerId);

    metrics.addMetadata('CartDetails', event.cartState);

    // ... rest of the code
  }
}

const myFunction = new Lambda();
export const handler = myFunction.handler.bind(myFunction);